src/cpu/aarch64/jit_brgemm_conv.cpp

```diff
@@ -0,0 +1,1491 @@
+
+    const auto M = is_ow_tail ? jcp.M_tail : jcp.M;
+    const auto kdh_l = kd_l * kh_l;
+    const auto ow_s = (kdh_l <= 0) ? ow : ker_ow_s;
+    const auto ow_f = (kdh_l <= 0) ? ow : ker_ow_f;
+    assert(ow <= ow_s && ow_s <= ow_f && ow_f <= ow + M);
+
+    brgemm_kernel_post_ops_t p;
+    if (do_postwork) {
+        p.ptr_bias = (void *)(bias_w);
+        p.ptr_scales = (void *)(&oscales[jcp.is_oc_scale * g_oc]);
+        p.ptr_binary_post_ops_rhs = post_ops_binary_rhs_arg_vec;
+        p.dst_orig = dst;
+    }
+
+    auto call_outwork_ker = [&](bool is_postwork, int ow_pw_s, int ow_pw_l) {
+        const auto outwork_ker = kernels_po_[get_ker_po_idx(ow_pw_l - 1,
+                                                     is_postwork, is_oc_tail)]
+                                         .get();
+        assert(ow_pw_l == outwork_ker->brg.bcast_dim);
+        if (is_postwork) {
+            p.ptr_out = dst_base
+                    + dst_dsz
+                            * (od * dst_h_sz + oh * dst_w_sz
+                                    + ow_pw_s * jcp.oc_without_padding);
+            p.ptr_in = static_cast<void *>(jcp.use_buffer
+                            ? (c_buffer + acc_dsz * (ow_pw_s - ow) * jcp.LDC)
+                            : p.ptr_out);
+        } else {
+            char *const ptr_Cz = jcp.use_buffer
+                    ? (c_buffer + acc_dsz * (ow_pw_s - ow) * jcp.LDC)
+                    : dst_base
+                            + dst_dsz
+                                    * (od * dst_h_sz + oh * dst_w_sz
+                                            + ow_pw_s * jcp.oc_without_padding);
+            p.ptr_out = static_cast<void *>(ptr_Cz);
+        }
+        (*outwork_ker)(&p);
+    };
+
+    if (ow < ow_s) {
+        // left side
+        const auto ow_pw_l = ow_s - ow;
+        if (do_init) call_outwork_ker(false, ow, ow_pw_l);
+        if (do_postwork) call_outwork_ker(true, ow, ow_pw_l);
+    }
+    if (ow_f < ow + M) {
+        // right side
+        const auto ow_pw_l = ow + M - ow_f;
+        if (do_init) call_outwork_ker(false, ow_f, ow_pw_l);
+        if (do_postwork) call_outwork_ker(true, ow_f, ow_pw_l);
+    }
+}
+
+template <cpu_isa_t isa>
+void brgemm_convolution_fwd_t<isa>::call_brgemm_kernel(brgemm_thread_ctx_t &btc,
+        int brg_idx, int batch_size, char *ptr_C, char *ptr_D,
+        const char *bias_w, int g_oc, bool do_postops,
+        const void *binary_post_ops_rhs) const {
+
+    const auto _pd = pd();
+    const auto &jcp = _pd->jcp_;
+
+    const auto brg_ker = brg_kernels_[brg_idx].get();
+
+    // TODO: avoid costly tile reconfigurations
+    if (is_amx) {
+        if (std::memcmp(btc.cur_palette.a, brg_kernel_palettes_[brg_idx].a,
+                    //AMX_PALETTE_SIZE)
+                    64)
+                != 0) {
+            //amx_tile_configure(brg_kernel_palettes_[brg_idx].a);
+            printf("%s, %d, unimplemented\n", __FILE__, __LINE__);
```

Comment:
we typically don't do printf inside code.

Reply:
I'm sorry about this.
I removed the printf.